test/utils.ts

```diff
@@ -68,8 +68,8 @@ export const isClose = (actual: number | bigint, expected: number | bigint, erro
   if (typeof actual !== typeof expected) return false
 
   if (typeof actual === 'bigint' && typeof expected === 'bigint') {
-    const upper = expected + BigInt(error),
-      lower = expected - BigInt(error)
+    const upper = Number(expected) + error,
```

Comment:
The changes here could potentially cause issues if the `expected` value is `typeof BigInt` and a value greater than can be represented as a javascript native Float64.

How would you feel about changing it to:
```ts
export const isClose = (
  actual: number | bigint,
  expected: number | bigint,
  error = 0.001,
  castToNum = false
) => {
  if (typeof actual !== typeof expected) return false

  if (typeof actual === 'bigint' && typeof expected === 'bigint' && !castToNum) {
    const upper = expected + BigInt(error),
      lower = expected - BigInt(error)
    if (actual < lower || actual > upper) {
      return false
    }
    return true
  } else if (typeof actual === 'bigint' && typeof expected === 'bigint' && castToNum) {
    const upper = Number(expected) + error,
      lower = Number(expected) - error
    if (actual < lower || actual > upper) {
      return false
    }
  } else if (typeof actual === 'number' && typeof expected === 'number') {
    const upper = (expected as number) + error,
      lower = expected - error
    if (actual < lower || actual > upper) {
      return false
    }
    return true
  }
}
```

That way we can pass in `castToNum = true` and opt in to this behavior?

Reply:
I think this was an unrelated change that got accidentally included. However, the issue was that `BigInt(error)` always gave me an error when `error` is not an integer (which it usually isn't).

Reply:
I will probably revert this change for this PR.

This should probably be fixed in a separate PR, but I'm not entirely sure what would be the best way to approach it.

Reply:
Ohhhh, good catch! I had hit that elsewhere in the code, this should fix it:

```ts
export const isClose = (actual: number | bigint, expected: number | bigint, error = 0.001) => {
  if (typeof actual !== typeof expected) return false

  if (typeof actual === 'bigint' && typeof expected === 'bigint') {

    const factor = 1 / error
    const upper = expected * BigInt(factor) + BigInt(error * factor),
      lower = expected * BigInt(factor) - BigInt(error * factor)
    if (actual < lower || actual > upper) {
      return false
    }
    return true
  } else if (typeof actual === 'number' && typeof expected === 'number') {
    const upper = (expected as number) + error,
      lower = expected - error
    if (actual < lower || actual > upper) {
      return false
    }
    return true
  }
}
```

Reply:
I've submitted a separate PR with the above fix, so disregard this feedback.